### ISSUE TITLE
Allow private tabs to use different sessions (partitions)

### ISSUE THREAD

**Issue by bsclifton** [brave/brave-browser#776]
Carried over from https://github.com/brave/browser-laptop/issues/597

Tabs in a private window (either by default or via configuration) should use a different partition

### Steps to reproduce
1. Open a new private window
2. Visit https://trac.torproject.org/projects/tor and login
3. Open a new private tab and visit Visit https://trac.torproject.org/projects/tor
4. Notice this new tab shows you as logged in (from account in step 2)

**Comment by tomlowenthal** [comment]
I don't think that this is directly transferrable from the private tabs to private windows model. I'd expect all the tabs in the same private window to share state. Obviously, that state shouldn't be shared with the regular browser and should be dropped when that window is closed.

My friendly amendment would be to allow the creation of multiple distinct private sessions. I feel like there's a fair bit of work to make those distinct from each other. I want to build this feature, but probably at the window level rather than at tab scope.

**Comment by allisonphillips** [comment]
Even when I open new private *windows* (not tabs), it is still picking up the session from other previously closed private windows. Not deleting session data for closed private windows kinda defeats the purpose of private browsing.

**Comment by tomlowenthal** [comment]
A private browsing session starts when you open your first private window, and ends when there are no private windows open. If you leave your private windows open, your session will persist until you close them.

**Comment by NumDeP** [comment]
Hi @tomlowenthal I have to say I agree with Allison.

I've noticed this for quite sometime but didn't go out of my way to test it.

Using incognito in Brave Linux (OpenSuse) and exiting the browser and later using Incognito for another session still displays a few recommended titles on yoututbe, recommended titels that are peculiar and not typical. I don't mean on the homepage of yt, though there have been instances where that has happened but on the right-hand list when viewing videos.

Take this for example. If you got a clean system and you open a clean browser, nothing at all saved and every aspect of history completely wiped from the beginning of time. You open open private window and you direct yourself to youtube and and type "Noam Chomsky" and peruse through a historical video, noting particularly new. Wouldn't you find it rather dubious if you closed every window and utilized a system cleaner and that same old Noam Chomsky video from way in the past shows up as a recommended video when you open a new private window and after having typed "funny videos" and that same Noam Chomsky video shows up on the right-hand side of the list, even if after restart of the PC.

**Comment by securitygeneration** [comment]
Wondering if there will be any movement on this one? Your idea @tomlowenthal, to begin with window-wide private sessions, makes sense to begin with (as the current limitation is really the lack of ability to have more than one private session - windowed or tabbed). Seeing as Brave is kinda geared towards privacy, this kind of feature would make sense. Safari seems to be the only browser that enforces separate sessions for all private browsing sessions (windows _and_ tabs).

**Comment by kspeakman** [comment]
I came here just to find if the idea that @tomlowenthal mentioned has suggested here. I do a bit of IT and web dev work and sometimes need to login to the same service with different credentials. For example, separate AWS accounts for different products. Or separate logins for personal, work, and community groups. I typically use a private window to have an isolated session with the same service. But in cases where I need a 3rd login to the same service, I have to open a different browser. It would be awesome to have each private window be its own session. Each tab being in the same session as its parent private window is desirable at times. For example when I need multiple AWS service pages open for the same account.
